rpc/src/rpc.rs

```diff
@@ -3645,8 +3645,11 @@ pub mod rpc_full {
             let (wire_transaction, unsanitized_tx) =
                 decode_and_deserialize::<VersionedTransaction>(data, binary_encoding)?;
 
-            let preflight_commitment =
-                preflight_commitment.map(|commitment| CommitmentConfig { commitment });
+            let preflight_commitment = if skip_preflight {
+                Some(CommitmentConfig::processed())
```

Comment:
on second thought, would `confirmed` be more reliable here?  more likely to pick up ephemeral forks with `processed`

Reply:
I can't see a drawback to picking up an ephemeral fork here. This is only being used to look for the recent_blockhash to determine last_valid_block_height. If the blockhash is from a confirmed or finalized block, it should appear in the ancestor tree of the fork.

Reply:
But I will hold off merging until you reply, in case you've thought of something I missed.

Reply:
was thinking something like...

```
C - 1' - 2'    - 3'(STS)
  \ 1  - 2(TX) - 3
```

Reply:
I mean, the bigger problem for the tx using a processed blockhash in that example is really that the transaction won't ever land if the leaders are on the same fork-prime as STS.
However, STS will still retry the transaction the fixed number of blocks (300) as per the logic at L3668

Reply:
...I'm going to merge this as-is, but there's no big risk in tweaking this later.

Reply:
I'll similarly land https://github.com/solana-labs/solana-web3.js/pull/2415. If you change your mind on the commitment, let's make the change in the client too!